solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -127,7 +127,7 @@ pub mod solana_ibc {
         ctx: Context<'a, 'a, 'a, 'info, InitEscrow<'info>>,
         port_id: ibc::PortId,
         channel_id_on_b: ibc::ChannelId,
-        base_denom: String,
+        hashed_base_denom: Vec<u8>,
```

Comment:
Why not fixed size hash? Also should this now be called `hashed_denom`?
```suggestion
        hashed_denom: CryptoHash,
```

Reply:
Right, but as per Slack discussion, don’t we want to include full denom?

Reply:
Yes i think we might need full denom. we might have to add trace prefix when we receive the packet since ibc seems to be removing.